docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,664 @@
+      - optional
+
+##### Preparing Credentials
+The admin user (and any users) need:
+  - a token or a password to identify them. 
+  - tokens are just long alphanumeric strings, e.g. 32 chars.  See
+    - `TOKEN=$(dd if=/dev/urandom bs=128 count=1 2>/dev/null | base64 | tr -d "=+/" | dd bs=32 count=1 2>/dev/null)`
+
+Any tokens and.or passwords need to stored in a file for the apiserver
+to read.  This guide uses `/var/lib/kube-apiserver/known_tokens.csv`.
+The format for this file is described in the [authentication documentation](
+../authentication.md).
+
+For distributing credentials to clients, the convention in Kubernetes is to put the credentials
+into a [kubeconfig file](../kubeconfig-file.md).
+
+The kubeconfig file for the administrator can be created as follows:
+ - If you have already used Kubernetes with a non-custom cluster, you will already have a `$HOME/.kube/config` file.
+ - You need to add certs, keys, and the master IP to the kubeconfig file:
+    - if using the firewall-only security option, set the apiserver this way:
+      - `kubectl config set-cluster $CLUSTER_NAME --server=http://$MASTER_IP --insecure-skip-tls-verify=true`
+    - otherwise, do this to set the apiserver ip, client certs, and user credentials.
+      - `kubectl config set-cluster $CLUSTER_NAME --certificate-authority=$CA_CERT --embed-certs=true --server=https://$MASTER_IP`
+      - `kubectl config set-credentials $CLUSTER_NAME --client-certificate=$CLI_CERT --client-key=$CLI_KEY --embed-certs=true --token=$TOKEN`
+    - Set your cluster as the default cluster to use:
+      - `kubectl config set-context $CLUSTER_NAME --cluster=$CLUSTER_NAME --user=admin`
+      - `kubectl config use-context $CONTEXT  --cluster=$CONTEXT`
+
+Next, make a kubeconfig file for the kubelets and kube-proxy.  There are a couple of options for how 
+many distinct files to make:
+  1. use the same credential as the admin
+    - simple
+    - harder to add finer grained permissions later
+  1. one token and kubeconfig file for all kubelets, one for all kube-proxy, one for admin.
+    - mirrors what is done on GCE today
+  1. different credentials for every kubelet, etc.
+    - we are working on this but all the pieces are not ready yet.
+
+You can make the files by copying the `$HOME/.kube/config`, by following the code
+in `cluster/gce/configure-vm.sh` or by using the following template:
+```
+apiVersion: v1
+kind: Config
+users:
+- name: kubelet
+  user:
+    token: ${KUBELET_TOKEN}
+clusters:
+- name: local
+  cluster:
+    certificate-authority-data: ${CA_CERT_BASE64_ENCODED}
+contexts:
+- context:
+    cluster: local
+    user: kubelet
+  name: service-account-context
+current-context: service-account-context
+```
+Put the kubeconfig(s) on every node.  The examples later in this
+guide assume kubeconfigs in `/var/lib/kube-proxy/kubeconfig` and
+`/var/lib/kubeletproxy/kubeconfig`.
+
+#### Selecting Binaries
+You will need binaries for:
+  - etcd
+  - docker or rkt
+  - kubelet
```

Comment:
Can we add buckets to the list (one for containers and another for k8s):

 - etcd
 - A container image, either:
     - Docker
     - rkt
 - Kuberenetes:
    - kubelet
    - ...